### DISCUSSION TITLE
error while generating pdf

### DISCUSSION THREAD

**Discussion by zsofiaj** [thegetty/quire#745] [Get Help] [answered]
I mentioned this under a different topic, but I thought I should make it a new thread. The most recent attempt at generating a pdf of our project failed, and I got an error message that I cannot decipher. I've pasted it below. Has anyone encountered this?

@geealbers I'm not sure if this has anything to do with those printImage values I added in figures.yaml, but I this is the first time I attempted to build a pdf since I did that. 

Thank you!

zsofiajilling@COENGL-02E1D2 italian-paintings-march2023 % quire pdf
[CLI:lib/pdf] generating PDF using Paged.js
[CLI:lib/pdf/pagedjs] printing /Users/zsofiajilling/italian-paintings-march2023/_site/pdf.html

  Puppeteer old Headless deprecation warning:
    In the near feature `headless: true` will default to the new Headless mode
    for Chrome instead of the old Headless implementation. For more
    information, please see https://developer.chrome.com/articles/new-headless/.
    Consider opting in early by passing `headless: "new"` to `puppeteer.launch()`
    If you encounter any bugs, please report them to https://github.com/puppeteer/puppeteer/issues/new/choose.

Error: Failed to launch the browser process!
assertion failed: emulated forward to an arm pc that isn't in translated code.  arm_pc=0x126960064 abi_kind=4 emulation_interval=[0x12454abdc,0x126953060) instruction_interval=[0x126953038, 0x126953060) x86_rip=0x113577398
(ThreadContextRegisterState.cpp:651 move_to_instruction_boundary)

TROUBLESHOOTING: https://pptr.dev/troubleshooting

    at Interface.onClose (file:///usr/local/lib/node_modules/@thegetty/quire-cli/node_modules/@puppeteer/browsers/lib/esm/launch.js:253:24)
    at Interface.emit (node:events:525:35)
    at Interface.close (node:internal/readline/interface:533:10)
    at Socket.onend (node:internal/readline/interface:259:10)
    at Socket.emit (node:events:525:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
zsofiajilling@COENGL-02E1D2 italian-paintings-march2023 %

**Comment by zsofiaj** [comment]
I realized I had a handful of images that were larger than we intended. I resized those, and now the pdf output fine. Could that have been the problem, or did the error above just magically resolve itself?

**Reply by geealbers** [reply]
This isn't an error I've seen before and is related to one of the dependencies for paged.js which generates the PDF file, rather than with Quire itself. I'm pinging @fchasen here in case it's something he's seen before. But I do think overly large images could have caused it. @zsofiaj do you remember how large the images were and how many of them there were? I'd like to do some testing on our end to confirm and then we can add a note to the Quire docs if that is indeed the cause.

**Reply by zsofiaj** [reply]
Thanks, @geealbers . There were about 5–10 images that were not resized properly, but they weren't huge. One was maybe 13 MB, and the rest 5–7 MB. All of the images are under 3 MB now, most under 2 MB. But there are around 175 images now, and many more coming, so the images alone are adding up to around 300 MB in the project. The project director folder now contains around 1.5 GB. I'm not completely sure how it adds up to that, but I've got around 300 MB in the content folder, 300 MB in the _site folder, and 300 MB in the _epub folder, 170 MB in the node_modules folder, and nothing else is significant. I'm not sure where that extra 500ish MB is hiding. I don't have a pdf in the project folder currently. 

I'm a little concerned the project is going to get too large to function properly. We still have many catalogue entries to add. Will that be a problem? The project is already too large to use the Netlify drop page to do a manual deploy.

**Comment by fchasen** [comment, marked as answer]
That headless message is just a warning (which I need to look into!) but the error here is `emulated forward to an arm pc that isn't in translated code.` which seems to be a Mac m1 / m2 arm error?

Can you try a few things:

1. Make sure you aren't running your terminal in Rosetta mode by opening `/System/Applications/Utilities` and right clicking on `Terminal`. `Open using Rosetta` should be **unchecked**.
2. Check your version of Node with `node -v`, it should be the current LTS `18.16.0`
3. Reinstall Quire and all it's dependencies with: `npm clean-install  --global @thegetty/quire-cli`

**Reply by zsofiaj** [reply]
Thanks so much, @fchasen , and apologies for the slow reply. I wasn't able to confirm if Terminal was running in Rosetta mode or not, but I updated Node, and that seemed to do the trick. I'm not on v20.4.0. 

I didn't reply earlier because the pdf build had started working again, but then I encountered other pdf issues, and we temporarily abandoned the pdf output. We're trying to get it working again, but I'm encountering other issues, so I'll post a new discussion topic in a moment and will ask @geealbers if he has any ideas.
